core/client/verify.go

```diff
@@ -12,59 +12,56 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-package kt
+package client
 
 import (
 	"context"
-	"crypto"
 	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
-	"log"
 
 	"github.com/google/keytransparency/core/crypto/commitments"
 	"github.com/google/keytransparency/core/crypto/vrf"
 	"github.com/google/keytransparency/core/mutator/entry"
-
 	"github.com/google/trillian"
 	"github.com/google/trillian/client"
-	"github.com/google/trillian/merkle"
-	"github.com/google/trillian/merkle/hashers"
 
 	pb "github.com/google/keytransparency/core/api/v1/keytransparency_proto"
-	tcrypto "github.com/google/trillian/crypto"
 )
 
 var (
 	// ErrNilProof occurs when the provided GetEntryResponse contains a nil proof.
 	ErrNilProof = errors.New("nil proof")
-
-	// Vlog is the verbose logger. By default it outputs to /dev/null.
-	Vlog = log.New(ioutil.Discard, "", 0)
 )
 
 // Verifier is a client helper library for verifying request and responses.
 type Verifier struct {
 	vrf         vrf.PublicKey
-	hasher      hashers.MapHasher
-	mapPubKey   crypto.PublicKey
+	mapVerifier *client.MapVerifier
 	logVerifier client.LogVerifier
 }
 
-// New creates a new instance of the client verifier.
-func New(vrf vrf.PublicKey,
-	hasher hashers.MapHasher,
-	mapPubKey crypto.PublicKey,
+// NewVerifier creates a new instance of the client verifier.
+func NewVerifier(vrf vrf.PublicKey,
+	mapVerifier *client.MapVerifier,
 	logVerifier client.LogVerifier) *Verifier {
 	return &Verifier{
 		vrf:         vrf,
-		hasher:      hasher,
-		mapPubKey:   mapPubKey,
+		mapVerifier: mapVerifier,
 		logVerifier: logVerifier,
 	}
 }
 
+// Index computes the index from a VRF proof.
+func (v *Verifier) Index(vrfProof []byte, domainID, appID, userID string) ([]byte, error) {
+	uid := vrf.UniqueID(userID, appID)
+	index, err := v.vrf.ProofToHash(uid, vrfProof)
+	if err != nil {
+		return nil, fmt.Errorf("vrf.ProofToHash(%v, %v): %v", appID, userID, err)
```

Comment:
This error message makes it look like appID and userID were passed to ProofToHash(), when that was only indirectly the case.

Reply:
done